game-core/src/main/java/games/strategy/debug/ErrorMessage.java

```diff
@@ -128,7 +129,14 @@ private void setUploadRecord(final LogRecord record) {
       }
       INSTANCE.uploadButton.addActionListener(e -> {
         hide();
-        StackTraceReportView.showWindow(windowReference, serviceClient, record);
+        if (serviceClient.canSubmitErrorReport()) {
+          StackTraceReportView.showWindow(windowReference, serviceClient, record);
+        } else {
+          SwingComponents.showDialog(
```

Comment:
This is a user facing change here. THe idea is if a user has hit the rate limit, we do not want them to fill in the optional details dialog to then click submit and get rejected then. Instead we check with the server when user clicks "report to triplea" and if they have hit the limit will show them a message instead of the error report form.

Reply:
This will probably break once a user (or 2 users with the same public IP) is getting 2 errors at once in 2 different windows when they're just about to hit the limit.

Not sure if it's worth dealing with this rare to unlikely case.

Reply:
Good consideration. It does seem like it should be pretty rare, I'd be a bit surprised if many people even hit the limit at all. We could increase the limit to 10 to see how it goes. On the plus side, as this is server side code, we can re-deploy with an updated limit pretty readily (so the risk of getting it wrong now is not that great).

Reply:
TLDR; IMO we can see how it goes for now, I'd be surprised if we were to see that problem.